storage/src/main/java/tech/pegasys/teku/storage/server/network/DatabaseNetwork.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.network;
+
+import static com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature.WRITE_DOC_START_MARKER;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+import tech.pegasys.teku.storage.server.DatabaseStorageException;
+import tech.pegasys.teku.util.config.Eth1Address;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DatabaseNetwork {
+  @JsonProperty("fork_version")
+  @VisibleForTesting
+  final String forkVersion;
+
+  @JsonProperty("deposit_contract")
+  @VisibleForTesting
+  final String depositContract;
+
+  @JsonCreator
+  DatabaseNetwork(
+      @JsonProperty("fork_version") final String forkVersion,
+      @JsonProperty("deposit_contract") final String depositContract) {
+    this.forkVersion = forkVersion;
+    this.depositContract = depositContract;
+  }
+
+  public static DatabaseNetwork init(
+      final File source, Bytes4 forkVersion, Eth1Address depositContract) throws IOException {
+    final String forkVersionString = forkVersion.toHexString().toLowerCase();
+    final String depositContractString =
+        depositContract == null ? "" : depositContract.toHexString().toLowerCase();
+    final ObjectMapper objectMapper =
+        new ObjectMapper(new YAMLFactory().disable(WRITE_DOC_START_MARKER));
+    if (source.exists()) {
+      final DatabaseNetwork databaseNetwork =
+          objectMapper.readerFor(DatabaseNetwork.class).readValue(source);
+
+      if (!databaseNetwork.forkVersion.equals(forkVersionString)) {
```

Comment:
nit: I'd flip this equals the other way around. We know for certain that `forkVersionString` isn't `null` but `databaseNetwork` just came from an external file.  We'll fail to start either way but it will be much more neat if we avoid `NullPointerException`.
```suggestion
      if (!forkVersionString.equals(databaseNetwork.forkVersion)) {
```